relayer/chains/wasm/provider.go

```diff
@@ -366,25 +368,34 @@ func (p *Provider) subscribeTxResult(ctx context.Context, tx *sdkTypes.TxRespons
 }
 
 func (p *Provider) MessageReceived(ctx context.Context, key *relayTypes.MessageKey) (bool, error) {
-	queryMsg := &types.QueryReceiptMsg{
-		GetReceipt: &types.GetReceiptMsg{
-			SrcNetwork: key.Src,
-			ConnSn:     key.Sn.String(),
-		},
-	}
-	rawQueryMsg, err := jsoniter.Marshal(queryMsg)
-	if err != nil {
-		return false, err
-	}
+	switch key.EventType {
+	case events.EmitMessage:
+		queryMsg := &types.QueryReceiptMsg{
+			GetReceipt: &types.GetReceiptMsg{
+				SrcNetwork: key.Src,
+				ConnSn:     key.Sn.String(),
+			},
+		}
+		rawQueryMsg, err := jsoniter.Marshal(queryMsg)
+		if err != nil {
+			return false, err
+		}
 
-	res, err := p.client.QuerySmartContract(ctx, p.cfg.Contracts[relayTypes.ConnectionContract], rawQueryMsg)
-	if err != nil {
-		p.logger.Error("failed to check if message is received: ", zap.Error(err))
-		return false, err
-	}
+		res, err := p.client.QuerySmartContract(ctx, p.cfg.Contracts[relayTypes.ConnectionContract], rawQueryMsg)
+		if err != nil {
+			p.logger.Error("failed to check if message is received: ", zap.Error(err))
+			return false, err
+		}
 
-	receiptMsgRes := types.QueryReceiptMsgResponse{}
-	return receiptMsgRes.Status, jsoniter.Unmarshal(res.Data, &receiptMsgRes.Status)
+		receiptMsgRes := types.QueryReceiptMsgResponse{}
+		return receiptMsgRes.Status, jsoniter.Unmarshal(res.Data, &receiptMsgRes.Status)
+	case events.CallMessage:
```

Comment:
Can't this default logic be handled in the relayer itself rather than each chain having to implement these cases just to return false,nil ?